CimSweep/ArtifactRetrieval/NetworkProfiles.ps1

```diff
@@ -0,0 +1,134 @@
+                
+            $Parameters = @{
+                Hive = 'HKLM'
+                SubKey = 'SOFTWARE\Microsoft\Windows NT\CurrentVersion\NetworkList\Profiles'
+            }
+    
+            Get-CSRegistryKey @Parameters @CommonArgs | ForEach-Object { 
+                
+                $ObjectProperties = @{}
+                $ObjectProperties['PSComputerName'] = $_.PSComputerName
+
+                Get-CSRegistryValue -Hive $_.Hive -SubKey $_.SubKey @CommonArgs | ForEach-Object { 
+                    
+                    $ValueName = $_.ValueName
+                    $ValueContent = $_.ValueContent
+
+                    switch ($ValueName) {
+                        
+                        { $_ -like "Date*" } {
+                            $BinaryReader = New-Object IO.BinaryReader (New-Object IO.MemoryStream (,$ValueContent))
+                        
+                            $Year = $BinaryReader.ReadInt16()
+                            $Month = $BinaryReader.ReadInt16()
+                            $null = $BinaryReader.ReadInt16() # skip week day
+                            $Day = $BinaryReader.ReadInt16()
+                            $Hour = $BinaryReader.ReadInt16()
+                            $Minute = $BinaryReader.ReadInt16()
+                            $Second = $BinaryReader.ReadInt16()
+                            $Millisecond = $BinaryReader.ReadInt16()
+                        
+                            $BinaryReader.BaseStream.Dispose()
+                            $BinaryReader.Dispose()
+
+                            $Date = New-Object datetime -ArgumentList @($Year, $Month, $Day, $Hour, $Minute, $Second, $Millisecond, 'Utc')
+                            
+                            $ObjectProperties[$ValueName] =  $Date.ToString('o')
+                        }
+                        
+                        'NameType' { 
+                            $ObjectProperties['Type'] = switch ($ValueContent) {
+                                      6 { 'Wired' }
+                                     23 { 'VPN' }
+                                     71 { 'Wireless' }
+                                default { $ValueContent }
+                            }
+                        }
+                        
+                        'Category' { 
+                            $ObjectProperties['Category'] = switch ($ValueContent) {
+                                0 { 'Public' }
+                                1 { 'Private' }
+                                2 { 'Domain' }
+                            }
+                        }
+                        
+                        'Managed' { $ObjectProperties['Managed'] = [bool]$ValueContent }
+                          
+                          default { $ObjectProperties[$ValueName] = $ValueContent }
+                    }
+                }
+                [PSCustomObject]$ObjectProperties
```

Comment:
ok so i looked into this again and based on my unit tests my stuff is processing things out.

With that said, i can certainly clean things up and make the path mean a bit more vs what shows up now.

![image](https://cloud.githubusercontent.com/assets/4265254/26565080/009951e0-44b6-11e7-98ad-e01fd296de56.png)